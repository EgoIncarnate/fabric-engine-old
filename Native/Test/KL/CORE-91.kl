struct Foo 
{ 
  Integer bar[]; 
}; 

function entry() 
{ 
    var Foo baz[]; 
    baz.resize(1); 
    baz[0].bar.push( 20 ); 
    report( baz );
} 
