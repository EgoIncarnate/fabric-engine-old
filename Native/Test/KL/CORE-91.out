GlobalList
  StructDecl( 'Foo' )
    StructDeclMember( 'bar', Integer[] )
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        StatementList
          VarDecl( 'baz', Foo[] )
        ExprStatement
          MethodOp( 'resize' )
            Var('baz')
            ArgList
              ConstSize( 1 )
        ExprStatement
          MethodOp( 'push' )
            StructMemberOp( 'bar' )
              IndexOp
                Var('baz')
                ConstSize( 0 )
            ArgList
              ConstSize( 20 )
        Report
          Var('baz')
[{bar:[20]}]
